tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +684,173 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+
+			DeleteAssets (project_path);
+			CopyAssets (project_path);
+
+			Clean (project_path);
+
+			DotNet.AssertBuild (project_path, GetDefaultProperties (runtimeIdentifiers));
+
+			var appExecutable = GetNativeExecutable (platform, appPath);
+			ExecuteWithMagicWordAndAssert (platform, runtimeIdentifiers, appExecutable);
+
+			var resourcesDirectory = GetResourcesDirectory (platform, appPath);
+
+			var assetsCar = Path.Combine (resourcesDirectory, "Assets.car");
+			Assert.That (assetsCar, Does.Exist, "Assets.car");
+
+			var doc = ProcessAssets (assetsCar);
+			Assert.IsNotNull (doc, "There was an issue processing the asset binary.");
+			var foundAssets = FindAssets (doc);
+
+			var TotalUniqueAssets = 18;
+			Assert.AreEqual (TotalUniqueAssets, foundAssets.Count, "There were a different number of assets found.");
+			Assert.IsFalse (foundAssets.Contains ("Data.DS_StoreDataTest"), "DS_Store files should not be included.");
+
+			var arm64txt = Path.Combine (resourcesDirectory, "arm64.txt");
+			var armtxt = Path.Combine (resourcesDirectory, "arm.txt");
+			var x64txt = Path.Combine (resourcesDirectory, "x64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm64")), File.Exists (arm64txt), "arm64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm")), File.Exists (armtxt), "arm.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-x64")), File.Exists (x64txt), "x64.txt");
+		}
+
+		void DeleteAssets (string project_path)
+		{
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+			try {
+				if (Directory.Exists (xcassetsDir))
+					Directory.Delete (xcassetsDir, true);
+			}
+			catch (Exception e){
+				Console.WriteLine($"Deleting Assets.xcassets failed: {e.Message}");
+			}
+		}
+
+		void CopyAssets (string project_path)
+		{
+			DirectoryInfo testingAssetsDir = new DirectoryInfo (Path.Combine (project_path, @"../../TestingAssets"));
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+
+			if (!testingAssetsDir.Exists)
+				throw new DirectoryNotFoundException ($"TestingAssets directory does not exist or could not be found: {testingAssetsDir}");
+
+			if (!Directory.Exists (xcassetsDir))
+				CopyDirectoryContents (testingAssetsDir, xcassetsDir);
+		}
+
+		void CopyDirectoryContents (DirectoryInfo dir, string DestDirectory) {
+			Directory.CreateDirectory (DestDirectory);
+			foreach (FileInfo file in dir.GetFiles ()) {
+				string tempPath = Path.Combine (DestDirectory, file.Name);
+				file.CopyTo (tempPath, false);
+			}
+			foreach (DirectoryInfo subDir in dir.GetDirectories ()){
+				CopyDirectoryContents (subDir, Path.Combine (DestDirectory, subDir.Name));
+			}
+		}
+
+		JsonDocument ProcessAssets (string assetsPath) {
+			try {
+				using (Process myProcess = new Process ()) {
```

Comment:
We have helper API to run a process and capture the output.

Something like this:

```csharp
var output = new StringBuilder ();
var executable = "xcrun";
var arguments = new string [] { "--sdk", "phones", "assetutil", "--info", assetPath };
var rv = Execution.RunWithStringBuildersAsync (executable, arguments, standardOutput: output, standardError: output, timeout: TimeSpan.FromSeconds (15)).Result;
Assert.AreEqual (0, rv.ExitCode, "ExitCode");

return JsonDocument.Parse (output);
```